### PR TITLE
DS-520 Use Typography components internally

### PR DESCRIPTION
**Pull request by gregaubert**

### CODE REVIEW

File: src/components/dropdown-menu/DropdownMenu.tsx [hunk 1]
```diff
@@ -154,19 +159,12 @@ const StyledHeaderLabelAndHelpText = styled.div`
   row-gap: var(---echoes-dimension-space-50);
 `;
 
-const StyledHeaderLabel = styled.div`
-  font: var(--echoes-typography-text-default-semi-bold);
-  overflow: hidden;
-  text-overflow: ellipsis;
-  white-space: nowrap;
+const StyledHeaderLabel = styled(Label)`
+  ${truncate}
```
Comment (jeremy-davis-sonarsource) [inline src/components/dropdown-menu/DropdownMenu.tsx:163]: :clap:

File: src/components/dropdown-menu/DropdownMenuItemBase.tsx [hunk 2]
```diff
@@ -127,7 +127,7 @@ const StyledIconCheck = styled(IconCheck)`
   font-size: var(--echoes-font-size-20);
 `;
 
-const CheckmarkPlaceholder = styled(StyledIconCheck)`
+const CheckmarkPlaceholder = styled(IconCheck)`
   visibility: hidden;
 `;
```
Comment (jeremy-davis-sonarsource) [inline src/components/dropdown-menu/DropdownMenuItemBase.tsx:132]: Why this change?

**Review by jeremy-davis-sonarsource — APPROVED** [review]
Nice!

File: src/components/dropdown-menu/DropdownMenuItemBase.tsx (hunk 2 above)
Comment (gregaubert) [reply src/components/dropdown-menu/DropdownMenuItemBase.tsx:132]: Nice catch, it should be reverted, I was testing if we really need the StyledIconCheck or not and we do.
